### PR TITLE
[MVP] Validate mint.json

### PR DESCRIPTION
**Pull request by Graviis**
# Summary

Validate every property in the `mint.json` so user can self generate their docs. We are going to return errors and warning to guide the users so they're capable to self host Mintlify

# Test Plan

Battery of tests for each validation method.

### CODE REVIEW

**Review by tobias-mintlify — COMMENTED** [review]
This looks great so far! If you want, you could make a PR and merge into `main` after addressing some of this feedback. Then, you could wrap up the `navigation` validation in a second PR. It's completely up to you whether you prefer several small PRs or one big one though.

File: packages/mint-validation/src/schemas/versions.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { z } from "zod";
+
+export const VersionsSchema = z.string().array().min(1, "Versions is missing");
```
[the remark this answers is not in the document]
Comment (Graviis) [reply packages/mint-validation/src/schemas/versions.ts:3]: Ok, so the navigation validation should raise an error when there's a version value there but not in this array, right?

File: packages/mint-validation/src/schemas/analytics.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        id: z.string({ required_error: "Pirsch id is missing" }),
+      })
+      .optional(),
+    posthog: z
+      .object({
+        apiKey: z.string({
+          required_error: "Posthog apiKey is missing",
+        }),
+        apiHost: z.string({
+          required_error: "Posthog apiHost is missing",
```
[the remark this answers is not in the document]
Comment (Graviis) [reply packages/mint-validation/src/schemas/analytics.ts]: Cool!

File: packages/mint-validation/src/schemas/versions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobias-mintlify) [reply packages/mint-validation/src/schemas/versions.ts]: Yes

**Review by tobias-mintlify — APPROVED** [review]
Looks great 🥳

**Pull request merged**